src/ipc/ipc4/helper.c

```diff
@@ -803,8 +803,11 @@ static int ipc4_update_comps_direction(struct ipc *ipc, uint32_t ppl_id)
 		if (icd->cd->direction_set)
 			continue;
 
+		if (list_is_empty(&icd->cd->bsource_list))
+			continue;
+
 		src_buf = list_first_item(&icd->cd->bsource_list, struct comp_buffer, sink_list);
-		if (src_buf && src_buf->source->direction_set) {
+		if (src_buf->source->direction_set) {
```

Comment:
no, `list_first_item()` won't return `NULL`, so the "list empty" test above is enough

Reply:
??
It will not return NULL, thats why I removed if (src_buf) NULL check part.

Reply:
@abonislawski ouch... I do apologise... almost looks like I'd been reviewing on a Monday morning, but it was a Wednesday... weird, sorry